dissect/update/__init__.py

```diff
@@ -0,0 +1,220 @@
+    # from that file inside subprocess.run calls.
+    if args.online:
+        args.file = PYPROJECT_ONLINE_URL
+        log.info(f"The following url will be used to determine dependencies: {args.file}")
+        try:
+            input("Press ENTER to continue..")
+        except KeyboardInterrupt:
+            print()
+            return
+
+    # We attempt to obtain our dependencies from a pyproject.toml file.
+    pyproject = load_pyproject_toml(args.file)
+
+    # We check if the current environment has any git editable install locations.
+    editable_installs = list(find_editable_installs(args.verbose))
+
+    if not pyproject and not editable_installs:
+        log.error("No pyproject.toml or editable installs found, exiting..")
+        return
+
+    if pyproject:
+        modules = pyproject["project"]["dependencies"]
+        log.info(f"Found {str(len(modules))} dependencies")
+
+        for module in modules:
+            pretty_module_name = module.split(">")[0].split("=")[0]
+
+            # If this module is also in the editable installs we found we skip them here.
+            if pretty_module_name in [m.get("name") for m in editable_installs]:
+                log.debug(f"Not updating module {pretty_module_name} as it is installed as editable")
+                continue
+
+            log.info(f"Updating dependency using pip: {pretty_module_name}")
+            # --pre does not do anything if pyproject.toml defines its dependencies strict like foo==1.0.0,
+            # so this only affects loose custom dependency definitions with, e.g. foo>1.0.0,<2.0.0.
+            _run(f"pip install '{module.strip(',')}' --upgrade --no-cache-dir --pre", args.verbose)
+
+    if editable_installs:
+        log.info(f"Found {str(len(editable_installs))} editable installs in current environment")
+
+        for module in editable_installs:
+            module_name = module.get("name")
+            module_path = module.get("editable_project_location")
+            log.info(f"Updating local dependency: {module_name} @ {module_path}")
+            # We assume that this is a git repository and we have git available to us.
+            _run(f"cd {module_path} && git pull && pip install -e .", args.verbose)
+
+    log.info("Finished updating dependencies")
+
+    # Display the version differences between the dependencies.
+    current_modules = environment_modules(args.verbose)
+    if initial_modules and current_modules:
+        for module in current_modules:
+            previous_module = next(filter(lambda prev_module: prev_module["name"] == module["name"], initial_modules))
+            module_name = module.get("name")
+            previous_version = previous_module.get("version")
+            current_version = module.get("version")
+
+            if previous_version != current_version:
+                print(f"{module_name} \x1b[31m{previous_version}\x1b[0m -> \x1b[32m\x1b[1m{current_version}\x1b[0m")
```

Comment:
Can you format this with some constants for the color codes? And maybe respect `NO_COLOR`?